src/Plugin/Field/FieldFormatter/InstagramEmbedFormatter.php

```diff
@@ -34,10 +38,16 @@ public function viewElements(FieldItemListInterface $items, $langcode) {
     foreach ($items as $delta => $item) {
       $matches = [];
       foreach (Instagram::$validationRegexp as $pattern => $key) {
-        if (preg_match($pattern, $item->value, $matches)) {
+        if (preg_match($pattern, $this->getEmbedCode($item), $item_matches)) {
+          $matches[] = $item_matches;
           break;
         }
       }
+
+      if (!empty($matches)) {
+        $matches = reset($matches);
```

Comment:
This is strange. If "shortcode" is missing then it would be missing also in current approach. Can you push the code that you have so I can check it?